### DISCUSSION TITLE
Streams are 4:3 after changing from rtsp to http

### DISCUSSION THREAD

**Discussion by micogg** [blakeblackshear/frigate#2647] [Ask A Question]
Hi,

To resolve a previous issue, I changed my streams from rtsp to http.
After changing to http and adding some input args for the http stream, all three of my cameras went from 16:9 to 4:3.
All cameras, recordings and birdseye are in a 4:3 aspect ratio.
I have only found one thread on fixing the 4:3 aspect ratio but I did not have any luck with it.  

I'm running the main Frigate instance in a separate container from HA, but not sure that matters here.
The 3 cameras are the troubled Reolink RLC-410's.
The main stream for all cameras is set to 2560x1920.
The sub stream for all cameras is set to 640x480.

record:
  enabled: True
  retain_days: 0
  events:
    objects:
      - person
    retain:
      default: 10
      objects:
        person: 15
detect:
  width: 640
  height: 480
  fps: 7
snapshots:
  enabled: True
  timestamp: True
  retain:
    default: 10
    objects:
      person: 15
birdseye:
  enabled: True
  width: 1280
  height: 720
  quality: 16
  mode: continuous
rtmp:
  enabled: False
ffmpeg:
  input_args:
    - -c:v
    - h264_cuvid
    - -avoid_negative_ts
    - make_zero
    - -fflags
    - nobuffer+genpts+discardcorrupt
    - -flags
    - low_delay
    - -strict
    - experimental
    - -analyzeduration
    - 1000M
    - 1000M
    - -probesize
    - 1000M
    - -rw_timeout
    - "5000000"
  output_args:
    detect: -f rawvideo -pix_fmt yuv420p
    record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
    rtmp: -c copy -f flv
cameras:
  back:
    ffmpeg:
      inputs:
        - path: http://192.168.1.154/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=user&password=password
          roles:
            - record
            - rtmp
      inputs:
        - path: http://192.168.1.154/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=user&password=password
          roles:
            - detect
    detect:
      enabled: False
  front left:
    ffmpeg:
      inputs:
        - path: http://192.168.1.175/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=user&password=password
          roles:
            - record
            - rtmp
      inputs:
        - path: http://192.168.1.175/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=user&password=password
          roles:
            - detect
    detect:
      enabled: False
    motion:
      mask:
        - 611,165,1280,215,1280,0,0,0,0,187
        - 354,160,0,167,0,438
  front right:
    ffmpeg:
      inputs:
        - path: http://192.168.1.153/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=user&password=password
          roles:
            - record
            - rtmp
      inputs:
        - path: http://192.168.1.153/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=user&password=password
          roles:
            - detect
    detect:
      enabled: True
    motion:
      mask:
        - 652,148,1280,179,1280,0,0,0,0,181
        - 1083,162,1280,178,1280,284
    zones:
      right_yard:
        coordinates: 0,720,1280,720,1280,267,1236,247,886,235,573,237,0,252
        objects:
          - person
          - car
          - motorcycle
<img width="1164" alt="Screen Shot 2022-01-14 at 2 14 25 PM" src="https://user-images.githubusercontent.com/33884470/149580211-031bed14-f8ea-40fb-bcec-55ee9f42d2ec.png">
<img width="1166" alt="Screen Shot 2022-01-14 at 2 14 09 PM" src="https://user-images.githubusercontent.com/33884470/149580215-71215fbf-568a-4f8b-a75a-3ffcc7d8c8a0.png">

**Comment by blakeblackshear** [comment]
Did you open them with VLC? Maybe that's just how the http streams come from the camera.

**Reply by micogg** [reply]
I did check with VLC and realized that the stream on the camera was set to a 4:3 ratio.  I went ahead and changed the camera setting for the below input to 2560x1440 which is a 16:9 aspect ratio. 

```
inputs:
  - path: http://192.168.1.153/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=user&password=password
  roles:
    - record
    - rtmp

```
VLC is showing 2560x1440 now for a 16x9 ratio when I look at the codec but Frigate is still streaming 4:3.  
Are there ffmpeg arguments I may be missing?

<img width="510" alt="Screen Shot 2022-01-19 at 12 00 53 PM" src="https://user-images.githubusercontent.com/33884470/150189168-1be5f369-ed53-489e-9830-4d00f5dff242.png">
<img width="1168" alt="Screen Shot 2022-01-19 at 12 13 31 PM" src="https://user-images.githubusercontent.com/33884470/150189867-865faa54-9fca-4082-8aa0-79d48dc277a4.png">

**Reply by blakeblackshear** [reply]
Birdseye uses the stream with the detect role. Are those 4:3?

**Reply by micogg** [reply]
Yes, thank you.  That fixed it! I set the detect role to the sub stream 4:3 resolution ( 640 x 480 ) previously without realizing that.  When I changed the detect settings to the main stream 16:9 resolution, it fixed the Birdseye view as well as the main camera view and recordings.  I did not realize it would affect all the views.  Thank you!

**Reply by DataBitz** [reply]
Won't this mean now that your detect is having to process the full resolution video, instead of the lower resolution substream?

**Comment by Davy-H** [comment]
Hi @micogg,

I've also got the RLC-410's but looking at the reolink website (https://support.reolink.com/hc/en-us/articles/900000441103-What-is-the-Aspect-Ratio-of-my-Camera)  they are support to be 4:3 cameras not 16:9. My RLC-410W's are 16:9 and display as wide screen in Frigate.  I would assume by changing the resolution on the cameras you are cutting out pixels or distorting the view.

Also note the substream resolution for the RLC410's are 640x480 (4:3) vs the RLC410W's which are 640:352 (16:9)
